### ISSUE TITLE
Incorrect prop validation when using this.props in constructor

### ISSUE THREAD

**Issue by gaearon** [yannickcr/eslint-plugin-react#259]
This is exactly like https://github.com/yannickcr/eslint-plugin-react/issues/254, but when using `this.props` instead of `props` argument in constructor.

```js
class TodoTextInput extends Component {
  constructor(props, context) {
    super(props, context);
    this.state = { text: this.props.text };
  }
}
TodoTextInput.propTypes = {
  text: PropTypes.string.isRequired
}
```

will give you `'props' is missing in props validation for TodoTextInput`.

**Comment by gaearon** [comment]
I added failing test cases in https://github.com/yannickcr/eslint-plugin-react/pull/260.

**Comment by yannickcr** [comment]
Thanks for the test case, I'll have a look.

**Comment by gaearon** [comment]
Thanks for a quick fix!
